YUViewLib/src/statistics/StatisticsData.cpp

```diff
@@ -165,6 +225,60 @@ QStringPairList StatisticsData::getValuesAt(const QPoint &pos) const
       }
     }
 
+    for (const auto &affineTFItem : this->frameCache.at(it->typeID).affineTFData)
+    {
+      const auto rect = QRect(
+          affineTFItem.pos[0], affineTFItem.pos[1], affineTFItem.size[0], affineTFItem.size[1]);
+      if (rect.contains(pos))
+      {
+        for( unsigned i=0; i<3; i++)
+        {
+          float vectorValue1, vectorValue2;
+          vectorValue1 = float(affineTFItem.point[i].first / it->vectorScale);
+          vectorValue2 = float(affineTFItem.point[i].second / it->vectorScale);
+          valueList.append(
+              QStringPair(QString("%1_%2[x]").arg(it->typeName).arg(i), QString::number(vectorValue1)));
+          valueList.append(
+              QStringPair(QString("%1_%2[y]").arg(it->typeName).arg(i), QString::number(vectorValue2)));
+        }
+        foundStats = true;
+      }
+    }
+
+    for (const auto &valueItem : this->frameCache.at(it->typeID).polygonValueData)
+    {
+      if (!valueItem.corners.size()) continue;
+      auto vectorPoly          = convertToQPointVectorPolygon(valueItem.corners);
+      if (stats::containsPoint( vectorPoly, pos, Qt::OddEvenFill))
+      {
+        int  value  = valueItem.value;
+        auto valTxt = it->getValueTxt(value);
+        valueList.append(QStringPair(it->typeName, valTxt));
+        foundStats = true;
+      }
+    }
+
+    for (const auto &polygonVectorItem : this->frameCache.at(it->typeID).polygonVectorData)
+    {
+      if (!polygonVectorItem.corners.size()) continue;
+      auto vectorPoly          = convertToQPointVectorPolygon(polygonVectorItem.corners);
+      if (stats::containsPoint( vectorPoly, pos, Qt::OddEvenFill))
+      {
+        if (it->renderVectorData)
+        {
+          float vx, vy;
+          // The length of the vector
+          vx = (float)polygonVectorItem.point.first / it->vectorScale;
+          vy = (float)polygonVectorItem.point.second / it->vectorScale;
+          valueList.append(
+              QStringPair(QString("%1[x]").arg(it->typeName), QString::number(vx)));
+          valueList.append(
+              QStringPair(QString("%1[y]").arg(it->typeName), QString::number(vy)));
+          foundStats = true;
+        }
+      }
+    }
```

Comment:
We should probably also add a test for this. Do you have some stats file that contains this data?

Reply:
Yes, here are some.  Skateboard_QP22_grepMV.vtmbmsstats contains among others AffineMVL0/1. Skateboard_QP22_grepGeo.vtmbmsstats has some geo statistics that use the polygons. There are some gaps in the statistics, since they are taken from my work on geometry padding (which is not coded, thus no statistics). The statistics themselves are the standard output by VTM though.

[Skateboard_QP22.zip](https://github.com/IENT/YUView/files/6334150/Skateboard_QP22.zip)